packages/next/src/server/base-server.ts

```diff
@@ -2733,55 +2732,103 @@ export default abstract class Server<
           }
 
           if (isPagesRouteModule(routeModule)) {
-            // Due to the way we pass data by mutating `renderOpts`, we can't extend
-            // the object here but only updating its `clientReferenceManifest` and
-            // `nextFontManifest` properties.
-            // https://github.com/vercel/next.js/blob/df7cbd904c3bd85f399d1ce90680c0ecf92d2752/packages/next/server/render.tsx#L947-L952
-            renderOpts.nextFontManifest = this.nextFontManifest
-            renderOpts.clientReferenceManifest =
-              components.clientReferenceManifest
+            const request: IncomingMessage = isNodeNextRequest(req)
+              ? req.originalRequest
+              : (req as any as IncomingMessage)
 
-            const request = isNodeNextRequest(req) ? req.originalRequest : req
             const response = isNodeNextResponse(res)
-              ? res.originalResponse
-              : res
+              ? (res.originalResponse as any as ServerResponse)
+              : (res as any as ServerResponse)
```

Comment:
Can we avoid the use of `as any` here? The `isNodeNextResponse` was used to type cast in the first place.

Reply:
Updated